code/game/area/Space Station 13 areas.dm

```diff
@@ -544,10 +544,333 @@ var/list/ghostteleportlocs = list()
 	icon_state = "syndie-elite"
 
 /area/syndicate_depot
-	name = "\improper Suspicious Supply Depot"
+	name = "Suspicious Supply Depot"
 	icon_state = "red"
 	tele_proof = 1
-	requires_power = 0
+
+	var/local_alarm = FALSE // Level 1: Local alarm tripped, bot spawned, red fire overlay activated
+	var/called_backup = FALSE // Level 2: Remote alarm tripped. Bot may path through depot. Backup spawned.
+	var/used_self_destruct = FALSE // Level 3: Self destruct activated. Depot will be destroyed shortly.
+
+	var/run_started = FALSE
+	var/run_finished = FALSE
+	var/list/hostiles = list()
+
+	var/list/guards_spawned = list()
+	var/used_lockdown = FALSE
+	var/destroyed = FALSE
+	var/something_looted = FALSE
+	var/on_peaceful = FALSE
+	var/list/peaceful_visitors = list()
+	var/peace_betrayed = FALSE
+	var/detected_mech = FALSE
+	var/obj/machinery/computer/syndicate_depot/syndiecomms/comms_computer = null
+	var/obj/structure/fusionreactor/reactor
+	var/list/hostiles_dead = list()
+	var/list/deployed_shields = list()
+	var/list/alert_log = list()

```

Comment:
This area has a lot of held references; it's kinda ick because it means when these objects are deleted, they won't garbage collect, as they'll be held in this list. Please make sure references are cleared so things garbage collect.

Reply:
/obj/structure/fusionreactor/Destroy() added.

I'm not sure how to handle hostiles_dead. On the one hand I do want it to be a list of /mobs, because its used to track who the mobs have seen die. On the other hand, I'm not sure how I can automatically remove /mobs from this list when they're deleted. Is there a known-good way to do that? or should I just have a timer that, once ever so often, deletes references to mobs that don't exist anymore?

deployed_shields automatically clears itself completely on proc/shields_down()

alert_log does not contain any references, it is simply a set of strings.